### ISSUE TITLE
"Golden axe" & " Tough Turf" issues

### ISSUE THREAD

**Issue by kaloun34** [jotego/jts16#16]
Hello
Since the update of 09/072022, the mra "Golden Axe" and "Tough Turf" switch to service mode at startup and it's impossible to leave it.
I managed it by forcing the service mode from the osd, but on reboot, same issue

**Comment by jotego** [comment]
Please try deleting your old config files.

**Comment by kaloun34** [comment]
I already tried it

**Comment by jotego** [comment]
Thank you. I'll check it.

**Comment by Videodr0me** [comment]
Same here. I think this can (also) be fixed by allowing to assign the "test button". This way one could just leave the "test program" and/or play around with it.

**Comment by kaloun34** [comment]
It's the MRA that's the problem.
Issue with the MRA "Golden Axe (set 6, US) (8751 317-123A)"
No issue with the alternatives MRAs
 -  Golden Axe (set 4, Japan) (FD1094 317-0121)
 -  Golden Axe (set 5, US) (FD1094 317-0122)

**Comment by asturur** [comment]
I want to add that the intro in goldenaxe has now broken graphic and sprites when the character are introduce.
The name is covered in white rectangles and the drawing seems off in colors.

(to leave the service mode, enter mister menu, press service mode on/off and you are out, that menu item serve as service button appartently)

**Comment by NewAstroCitys** [comment]
> I want to add that the intro in goldenaxe has now broken graphic and sprites when the character are introduce.
> The name is covered in white rectangles and the drawing seems off in colors.
> 
I have also experienced the same issue. Even though its not a game breaker it would be nice if this could be fixed.

**Comment by shuffle5** [comment]
i get this too (service menu, plus the attract mode is corrupted where it shows the characters drawings. )
![IMG_2023](https://user-images.githubusercontent.com/73855651/192160547-02636b8f-a1a8-4678-9b8d-9d7b51c64825.jpg)

**Comment by jotego** [comment]
verified in 7d64dbb
previous published version 21574e8 displays graphics correctly but it also starts in test mode. Versions earlier than that had the MCU version in a different RBF. 
I think these are two different problems. I take the test-mode bug to issue #19

**Comment by jotego** [comment]
Regression error introduced in ffb5179

```
 always @(posedge clk) if( pxl_cen ) begin
-    lyr0 <= tile_or_obj( obj_g[9:0], {4'd0, char_g[5:0] }, char_g[ 6], obj_prio==2'd3 );
-    lyr1 <= tile_or_obj( obj_g[9:0],        scr1_g[9:0]  , scr1_g[10], obj_prio>=2'd2 );
-    lyr2 <= tile_or_obj( obj_g[9:0],        scr2_g[9:0]  , scr2_g[10], obj_prio>=2'd1 );
-    lyr3 <= tile_or_obj( obj_g[9:0], {scr2_g[9:3], 3'd0 },       1'b0, 1'b1           );
+    lyr0 <= { 2'd0, tile_or_obj( obj_g[9:0], {4'd0, char_g[5:0] }, char_g[ 6], obj_prio==2'd3 ) };
+    lyr1 <= { 2'd1, tile_or_obj( obj_g[9:0],        scr1_g[9:0]  , scr1_g[10], obj_prio>=2'd2 ) };
+    lyr2 <= { 2'd2, tile_or_obj( obj_g[9:0],        scr2_g[9:0]  , scr2_g[10], obj_prio>=2'd1 ) };
+    lyr3 <= { 2'd2, tile_or_obj( obj_g[9:0], {scr2_g[9:3], 3'd0 },       1'b0, 1'b1           ) };
 end
 
 always @(*) begin
-    { shadow, pal_addr } =
+    { shadow, lyrsel, pal_addr } =
                (lyr0[10] ? lyr0[3:0]!=0 : lyr0[2:0]!=0) ? lyr0 : (
                (lyr1[10] ? lyr1[3:0]!=0 : lyr1[2:0]!=0) ? lyr1 : (
                (lyr2[10] ? lyr2[3:0]!=0 : lyr2[2:0]!=0) ? lyr2 : (
                 lyr3 )));
``` 
The shadow bit was incorrectly assigned because the newly layer-selection bits took the MSB part of the layer signals.

Fixed in d1ec409
